docs/pages/versions/unversioned/guides/using-nextjs.md

```diff
@@ -2,95 +2,237 @@
+> Warning: Support for Next.js is experimental. Please open an issue at [expo-cli/issues](https://github.com/expo/expo-cli/issues) if you encountered any problems.
 
-> Warning: Support for Next.js is experimental. Please open an issue at https://github.com/expo/expo-cli/ if you encountered any problem.
+[Next.js](https://nextjs.org/) is a React framework that provides simple page-based routing as well as server-side rendering. To use Next.js with Expo for web you can use a library called [`@expo/next-adapter`][next-adapter] which simplifies all of the config and setup required for using Next.js with Expo.
 
-## Using Next.js to your Expo for Web project
+Using Expo with Next.js means you can share all of your existing components and APIs across your mobile and web. Next.js has it's own Webpack config so **you'll need to start your web projects with the `next-cli` and not with `expo start:web`.**
 
-### Install and configure Next.js
+Next.js can only be used with Expo for web, this doesn't provide SSR for native apps.
 
-- Follow the [Running in the Browser](../../guides/running-in-the-browser) guide to add web support to your Expo project.
-- In your project, install Next.js: `npm i next --save`. (You may also have to run `npm i @babel/plugin-proposal-decorators --save-dev` due to an incompatibility.)
-- You will also need to add `expo-cli` as one of the `devDependencies` (`npm i expo-cli --save-dev`) if you want to build the project on a service such as [ZEIT Now](https://zeit.co/now).
-- Open `app.json` and set `expo.web.use` to `"nextjs"`:
+- [🏁 Setup](#-setup)
+  * [Expo projects with Next.js](#expo-projects-with-nextjs)
+  * [Next.js projects with Expo](#nextjs-projects-with-expo)
+  * [Shared steps](#shared-steps)
+  * [Offline support](#offline-support)
+  * [Using a custom server](#using-a-custom-server)
+  * [Handle server requests](#handle-server-requests)
+  * [Web push notifications support](#web-push-notifications-support)
+- [API](#api)
+  * [Config](#config)
+    + [`withExpo`](#withexpo)
+  * [Document](#document)
+    + [Customizing the Document](#customizing-the-document)
+  * [Server](#server)
+    + [`startServerAsync`](#startserverasync)
+    + [`createServerAsync`](#createserverasync)
+    + [`handleRequest`](#handlerequest)
+- [Limitations](#limitations-or-differences-comparing-to-the-default-expo-for-web)
+- [Contributing](#contributing)
 
-```json
-{
-  "expo": {
-    "web": {
-      "use": "nextjs"
-    }
-  }
-}
-```
+## 🏁 Setup
 
-- Since Next.js use page-based routing, your homepage will be `pages/index.js`, which is different from Expo project's main file (which is by default `App.js`). To support both native (mobile) app and Next.js, you could set `App.js` to:
+### Expo projects with Next.js
 
-```javascript
-import Index from './pages/index';
-export default Index;
-```
+Using SSR for web in your universal project. This is preferred because you'll get full access to both tools features.
```

Comment:
```suggestion
In this approach you would be using SSR for web in your universal project. This is the recommended path because it gives you full access to the features of Expo and Next.js.
```